### PR TITLE
Remove go-lxc dependencies

### PR DESCRIPTION
**Pull request by masnax**
Reimplements SetupTrust from LXD so we don't import too much from LXD. I think this is a simple enough function that we can duplicate it, not sure if we should bloat LXD's shared package with it.

### CODE REVIEW

File: microcloud/service/lxd_join.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		return fmt.Errorf("Failed generating trust certificate: %w", err)
+	}
+
+	post := api.CertificatesPost{
+		CertificatePut: cert.CertificatePut,
+		Password:       targetPassword,
+	}
+
+	err = target.CreateCertificate(post)
+	if err != nil && err.Error() != ErrCertificateExists.Error() {
```
Comment (tomponline) [inline microcloud/service/lxd_join.go:44]: Does the lxd server response when certificate already exists use a specific error code as ideally we would detect it using api.StatusErrorCheck() here.

**Review by tomponline — CHANGES_REQUESTED** [review]
Pr description is a little confusing as go-lxc is the go wrapper of liblxc not lxd's go client :)
Or is this an indirect dependency you're trying to avoid?

File: microcloud/service/lxd_join.go (hunk 1 above)
Comment (masnax) [reply microcloud/service/lxd_join.go:44]: Actually, looking into this more it looks like this error is long gone.

**Comment by masnax** [comment]
> Pr description is a little confusing as go-lxc is the go wrapper of liblxc not lxd's go client :) Or is this an indirect dependency you're trying to avoid? Does this also need a go mod update?

Yeah I'm trying to avoid bringing in liblxc.

File: microcloud/service/lxd_join.go (hunk 1 above)
Comment (masnax) [reply microcloud/service/lxd_join.go:44]: Actually not, apparently I was grepping incorrectly :)

Comment (masnax) [reply microcloud/service/lxd_join.go:44]: OK, I've updated the error to use `409 Conflict` if a certificate already exists.

**Comment by tomponline** [comment]
Thanks, ive merged the lxd pr now.

**Comment by stgraber** [comment]
@masnax I take it this needs an update for the LXD change as well as including a gomod update?

**Comment by masnax** [comment]
@stgraber I've updated this now for the LXD changes. Also since the LXD changes include the new init types, I've gone ahead and added a commit changing the initialization to use the `lxd init --preseed` command, as discussed with @tomponline

**Review by tomponline — APPROVED** [review]
Lgtm

**Comment by tomponline** [comment]
Good to see all of those indirect dependencies removed.

**Comment by masnax** [comment]
Just fixed a small typo (The preseed yaml was both the last argument and being written to stdin)

**Pull request merged**